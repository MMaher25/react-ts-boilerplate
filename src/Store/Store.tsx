import React, { createContext, useReducer } from 'react'
import { IState } from '../Interfaces/Interfaces'
import { reducer } from '../Reducers/Reducers'

export const initialState: IState = {
  pokemon: {
    abilities: [],
    base_experience: 0,
    forms: [],
    game_indices: [],
    height: 0,
    held_items: [],
    id: 0,
    is_default: false,
    location_area_encounters: '',
    moves: [],
    name: '',
    order: 0,
    species: [],
    sprites: [],
    stats: [],
    types: [],
    weight: 0,
  }
}

export const Store = createContext<IState | any>(initialState)

export const StoreProvider = ({ children }: JSX.ElementChildrenAttribute): JSX.Element => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return <Store.Provider value={{ state, dispatch }}>{children}</Store.Provider>
}
