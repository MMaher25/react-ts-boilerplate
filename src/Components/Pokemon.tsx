import React, { useContext, Fragment } from 'react'
import { Store } from '../Store/Store'
import { fetchDataAction } from '../Actions/Actions'

const Pokemon: React.FC = (): JSX.Element => {
  const { state, dispatch } = useContext(Store)
  const { pokemon } = state

  return (
    <div className="flex-item">
      <div>
        <img src={pokemon.sprites.front_default} />
      </div>
      <div>
        {pokemon.name.toUpperCase()} #{pokemon.id}
      </div>
      <button onClick={() => fetchDataAction(dispatch)}>
        Get Random Pokemon
      </button>
    </div>
  )
}

export default Pokemon