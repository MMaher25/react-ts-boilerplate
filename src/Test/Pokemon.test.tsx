import React from 'react';
import Pokemon from '../Components/Pokemon';
import { StoreProvider } from '../Store/Store';
import { create, act } from 'react-test-renderer';

describe("Pokemon component", () => {
  test("Matches the snapshot", () => {
    const pokemon = create(<StoreProvider><Pokemon /></StoreProvider>);
    expect(pokemon.toJSON()).toMatchSnapshot();
  });
});

describe("Pokemon component", () => {
  test("it shows a random Pokemon when clicked", async () => {
    const pokemon = create(<StoreProvider><Pokemon /></StoreProvider>);
    const instance = pokemon.root;
    const button = instance.findByType("button");
    const nameDiv = instance.findByProps({id: 'pokemon-name'});
    await act(() => button.props.onClick());
    act(() => console.log(nameDiv.props.children));
    expect(nameDiv.props.children[0]).not.toBe('');
    expect(nameDiv.props.children[2]).toBeGreaterThan(0);
    expect(nameDiv.props.children[2]).toBeLessThanOrEqual(807);
  });
});