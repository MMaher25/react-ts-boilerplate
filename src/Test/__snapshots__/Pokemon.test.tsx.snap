// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon component Matches the snapshot 1`] = `
<div
  className="flex-item"
>
  <div>
    <img
      id="image"
    />
  </div>
  <div
    id="pokemon-name"
  >
    
     #
    0
  </div>
  <button
    onClick={[Function]}
  >
    Get Random Pokemon
  </button>
</div>
`;
